src/kafka-net/BrokerRouter.cs

```diff
@@ -121,30 +123,60 @@ public List<Topic> GetTopicMetadata(params string[] topics)
         /// <param name="topics">List of topics to update metadata for.</param>
         /// <remarks>
         /// This method will ignore the cache and initiate a call to the kafka servers for all given topics, updating the cache with the resulting metadata.
-        /// Only call this method to force a metadata update.  For all other queries use <see cref="GetTopicMetadata"/> which uses cached values.
+        /// Only call this method to force a metadata update.  For all other queries use <see cref="GetTopicMetadataFromLocalCache"/> which uses cached values.
         /// </remarks>
-        public void RefreshTopicMetadata(params string[] topics)
+
+        public Task<bool> RefreshTopicMetadata(params string[] topics)
         {
-            //TODO need to remove lock here, try and move to lock free design
-            lock (_threadLock)
+            return RefreshTopicMetadata(_kafkaOptions.CacheExpiration, _kafkaOptions.RefreshMetadataTimeout, topics);
+        }
+
+        /// <summary>
+        /// refresh metadata Request get to server for all topic that there Cache Expire.
```

Comment:
please go over capitalizing
refresh -> Refresh
Cache -> cache
Expire -> expire

also that is not an english sentance